bitcoin/script/miniscript.h

```diff
@@ -780,7 +783,7 @@ struct Node {
     bool NeedsSignature() const { return GetType() << "s"_mst; }
 
     //! Do all sanity checks.
-    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize(); }
+    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize() && CheckTimeLocksMix(); }
```

Comment:
Now that timelock-safeness is just the "k" property, this could be simplified to just changing the required type to be `"Bmsk"_mst`. If `CheckTimeLocksMix` is independently interesting, it doesn't matter much.